cli/internal/runsummary/spaces.go

```diff
@@ -94,3 +102,34 @@ func newSpacesTaskPayload(taskSummary *TaskSummary) *spacesTask {
 		Logs:         string(taskSummary.GetLogs()),
 	}
 }
+
+// getStateOfRepo returns the sha and branch when in a git repo
+// Otherwise it should return empty strings right now.
+// We my add handling of other scms and non-git tracking in the future.
+func getStateOfRepo() (string, string) {
+	allEnvVars := env.GetEnvMap()
+
+	var sha string
+	var branch string
+
+	// If we're in CI, try to get the values we need from environment variables
+	if ci.IsCi() {
+		vendor := ci.Info()
+		branchVarName := vendor.BranchEnvVar
```

Comment:
agreed, i'm not sure how well this will go. I think the idea is to match vercel preview branches, but they're working with fundamentally different data (github) than us (git), so....